package/MDAnalysis/coordinates/TRR.py

```diff
@@ -151,6 +152,28 @@ class TRRReader(XDRBaseReader):
     _writer = TRRWriter
     _file = TRRFile
 
+    def _read_next_timestep(self, ts=None):
+        """copy next frame into timestep
+        
+        versionadded:: 2.4.0
+            TRRReader implements this method so that it can use
+            read_direct_xvf to read the data directly into the timestep
+            rather than copying it from a temporary array.
+        """
+        if self._frame == self.n_frames - 1:
```

Comment:
I'm not sure if this should get caught in `ProtoReader`... I'd be curious if it has coverage